grassroots-backend/src/grassroots-shared/cast.ts

```diff
@@ -0,0 +1,43 @@
+import { ClassConstructor, plainToInstance } from "class-transformer";
+import { validateSync } from "class-validator";
+
+// Needs to be imported somewhere for `enableImplicitConversion` to work.
+import "reflect-metadata";
+
+// Roughly, given a class, gives you an interface with its attributes.
+export type PropsOf<T> = {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
+  [K in keyof T as T[K] extends Function ? never : K]: T[K];
+};
+
+// Roughly, given a class, gives you an interface with the same attributes, but all untyped.
+export type PropsOfAsAny<T> = {
```

Comment:
would this be more accurately `PropsOfAsUnknown`?

Reply:
Whoops, thanks. Fixed.